resources/lib/ipwww_common.py

```diff
@@ -320,7 +311,6 @@ def StatusBBCiD():
 
 def CheckLogin(logged_in):
     if(logged_in == True or StatusBBCiD() == True):
-        logged_in = True
```

Comment:
Not sure I get this deletion. If logged_in is True, it gets overwritten (no harm done). If logged_in is False, but StatusBBCiD returns True, we are logged in, so logged_in should be set to True as well.

I wonder if a nested if would be more efficient. It would only call StatusBBCiD if logged_in is False, and therefore reduce the number of checks on the website.

Reply:
Parameter logged_in is local to the function and only used in the line above. Setting it to any value later on does indeed no harm, but has no effect either. So I thought it best to remove it.

Anyway, I've been researching the authentication procedures on the bbc website a bit, and I wonder if it is possible to just rely on automatically refreshed tokens, like a regular web browser does. That could make this whole function unnecessary. I hope to able to push some changes later today.

> I wonder if a nested if would be more efficient. It would only call StatusBBCiD if logged_in is False

That is in fact what is already happening. If logged_in is True the second part of the 'or' won't get evaluated.

Reply:
You are right. Guess I need to improve my Python skills.